src/Features/CSharp/Portable/InitializeParameter/InitializeParameterHelpers.cs

```diff
@@ -19,18 +19,30 @@ public static bool IsFunctionDeclaration(SyntaxNode node)
             || node is LocalFunctionStatementSyntax
             || node is AnonymousFunctionExpressionSyntax;
 
-        public static IBlockOperation GetBlockOperation(SyntaxNode functionDeclaration, SemanticModel semanticModel, CancellationToken cancellationToken)
+        public static bool TryGetBlockOperation(SyntaxNode functionDeclaration, SemanticModel semanticModel, CancellationToken cancellationToken, out IBlockOperation blockOperation)
         {
             var bodyOpt = GetBody(functionDeclaration);
             if (bodyOpt == null)
             {
-                return null;
+                blockOperation = null;
+                return true;
             }
 
             // body might be an expression in a lambda - in that case we wouldn't get a block operation out of that
-            return functionDeclaration is AnonymousFunctionExpressionSyntax
-                ? ((IAnonymousFunctionOperation)semanticModel.GetOperation(functionDeclaration, cancellationToken)).Body
-                : (IBlockOperation)semanticModel.GetOperation(bodyOpt, cancellationToken);
+            if (semanticModel.GetOperation(functionDeclaration, cancellationToken).Kind == OperationKind.AnonymousFunction)
+            {
+                blockOperation = ((IAnonymousFunctionOperation)semanticModel.GetOperation(functionDeclaration, cancellationToken)).Body;
+                return true;
+            }
+
+            if (semanticModel.GetOperation(bodyOpt, cancellationToken).Kind == OperationKind.Block)
+            {
+                blockOperation = (IBlockOperation)semanticModel.GetOperation(bodyOpt, cancellationToken);
+                return true;
+            }
```

Comment:
let's not call GetOperation redundantly :)   Instead of getting the operation, checking the kind, then getting the operation again, please call and store and use that value (this applied to both blocks).